addon/-private/system/snapshot.js

```diff
@@ -17,18 +17,23 @@ import { get } from '@ember/object';
 */
 export default class Snapshot {
   constructor(internalModel, options = {}) {
-    this._attributes = Object.create(null);
+    this.__attributes = null;
```

Comment:
I'd like (at some point) to start discussing moving some of these state caches into weakmaps now that we can use them indiscriminately...

Reply:
I can try and tackle that enhancement... maybe use a similar weakmap implementation found elsewhere in the code?